### DISCUSSION TITLE
choose a network interface to bind

### DISCUSSION THREAD

**Discussion by bart3nder** [v2fly/v2ray-core#2254] [General]
Hi!
Is there an option to choose a network interface to bind in our configs (Vmess/Vless/Trojan)? Like the way we do in shadowsocks with `-i` flag e.g. `ss-server -c config.json -i eth9`, which pass the traffic through an specified interface.

**Comment by slrslr** [comment]
I have similar question: How the V2ray config file should look like so i can use/bind VPN tunnel (Wireguard) located at interface 10.2.0.2 as my web browser socks proxy? (for example setting particular browser/app to proxy 127.0.0.1:1081 requests will go through the network interface 10.2.0.2)
I have found:
https://guide.v2fly.org/en_US/basics/routing/basics_routing.html
https://www.v2ray.com/en/configuration/protocols/socks.html
But really can not understand how to use it. Any example please?

**Comment by kingwilliam** [comment]
maybe use "sendthrough" can meet your request

https://www.v2fly.org/en_US/config/outbounds.html

**Reply by slrslr** [reply]
I have tried to replace 0.0.0.0 by 10.2.0.2 but that does not work. How the config.json file should look like please so it does what is described in my previous post or in OP post?

**Comment by AkinoKaede** [comment]
You can use `bindToDevice` setting for SocketSettings.

**Reply by slrslr** [reply]
Can you show the example full config file please?

**Reply by AkinoKaede** [reply]
Here is a simple config written in `JsonV5`.

```json
{
    "outbounds": [
        {
            "protocol": "freedom",
            "streamSettings": {
                "socketSettings": {
                    "bindToDevice": "eth0"
                }
            }
        }
    ]
}
```

**Reply by slrslr** [reply]
Which IP and port i will set in my app (for example web browser) so it uses this "outbounds" and utilize the device mentioned in your config for its network traffic? [I already have working and used socks /etc/v2ray/config.json](https://bin.disroot.org/?e60962223d0ff504#CJvtXEX1NpmyAoNx5t41R7Qw88WXRAxyF41osx9RQTEP), so i will just append your lines to it please?

**Reply by massood** [reply]
> Here is a simple config written in `JsonV5`.
> 

It is named `sockopt` now, not `socketSettings`:
```json
{
  "outbounds": [
    {
      "protocol": "freedom",
      "streamSettings": {
        "sockopt": {
          "bindToDevice": "eth0"
        }
      }
    }
  ]
}
```
